filters.go

```diff
@@ -169,6 +169,44 @@ func (p *Pipe) Join() *Pipe {
 	return Echo(output + terminator)
 }
 
+// Last reads from the pipe, and returns a new pipe containing only the last N
+// lines. If there is an error reading the pipe, the pipe's error status is also
+// set.
+func (p *Pipe) Last(lines int) *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	
+	defer p.Close()
+
+	if lines == 0 {
+		return NewPipe()
+	}
+
+	scanner := bufio.NewScanner(p.Reader)
+	input := make([]string, 0, lines)
+	for scanner.Scan() {
+		if len(input) == lines {
+			input = append(input[1:], scanner.Text())
```

Comment:
This seems like it might be a little inefficient; have you done any performance testing on it? I wonder if this might be a good place to use the [standard library](https://golang.org/pkg/container/ring/#pkg-overview) `ring` package?